src/components/checkbox/checkbox.tsx

```diff
@@ -134,4 +135,14 @@ export class Checkbox {
         this.change.emit(this.mdcCheckbox.checked);
         this.modified = true;
     }
+
+    @Watch('indeterminate')
+    private onIndeterminateChanged(newValue) {
+        // To properly remove the indeterminate state of the rendered
+        // input native control by the CheckboxTemplate, it's required
+        // to set indeterminate = false on the MDCCheckbox instance.
+        if (!newValue) {
```

Comment:
I think I'd prefer that we always set `this.mdcCheckbox.indeterminate`, rather than only set it when it becomes disabled. I know it means we'll have to do something about the horrendous animation when going from `indeterminate` to `checked`, but I don't much like the idea of having inconsistent state throughout the component.

Or am I being too pedantic? 🤔

Reply:
I mean… I _was_ ok with not touching `this.mdcCheckbox.indeterminate` at all, so adding this shouldn't really make a difference to me…

😖

Reply:
I only added the if-statement to avoid extra calls/renders but it also adds complexity to the code. I removed it now because it works just fine without it.

Reply:
Quoted from resolved thread:

> The fix I applied isn't pretty but it works. Instead of manually resetting the indeterminate state of the native element I added a watcher that unsets indeterminate on the MDCCheckbox instance instead _**(letting it do the job of toggling the state, which also unsets things like `aria-checked="mixed"`).**_

I just checked, and since we now update `this.mdcCheckbox.indeterminate` both when true and when false, MDC sets `aria-checked="mixed"` for us when indeterminate is true, not just when it's true from first render. So that's very good 👍

Reply:
I actually removed the comment, because it doesn't feel like we need to explain why we update the state of the MDC component. That's kind of the default, and there's more reason to comment on why if we _don't_ update it than if we _do_ 🙂

I also moved the watcher. We're a little inconsistent about where we place watchers. Originally I used to place all public functions first, then protected functions (which is only ever watchers), and last private functions. But Stencil's own guidelines say to place watchers immediately after the property they are watching. I don't like mixing the watchers and the props, as that makes it harder to see what props exist, but I have begun to place the watchers above the other functions, so they're at least close to the props they watch… I don't really know what's best to be honest. Consistency I guess, and consistent is the only thing we're not at the moment… 😣 well well…